backend/corpora/common/utils/dropbox.py

```diff
@@ -1,26 +1,32 @@
-from urllib.parse import urlparse, parse_qs, urlencode, urlunparse
-
+from urllib.parse import urlparse
 import regex
 import requests
 
+dropbox_link_rx = regex.compile(r"/s/[\w\d]+/.*")
```

Comment:
I think `\d` is a subset of `\w`, so you'd get the same effect with `\w+`. But, it doesn't include `-` or `.`, which would be valid, common characters in the url. Generally regex matching on urls is a little perilous.

More generally though, does dropbox make any promises about the structure of their shared links? Like if they all started with `/t/...` tomorrow, should that break the system?

Reply:
> Generally regex matching on urls is a little perilous.

I agree, and dropbox does not make any guarantee. Should we drop the regex all together. My goal here was to [prevent serverside request forgery](https://cheatsheetseries.owasp.org/cheatsheets/Server_Side_Request_Forgery_Prevention_Cheat_Sheet.html#available-protections) by matching against a regex as recommended by OWASP. Maybe the best thing to do it %encode the URL once we have it before sending to dropbox

> I think \d is a subset of \w, so you'd get the same effect with \w+. 

you're right \w includes everything.

> But, it doesn't include - or ., which would be valid, common characters in the url. 

No in thus particular part of the URL.

Reply:
After doing more investigating into SSRF prevention, strictly matching the netloc and schema seems sufficient. However I did find an interesting wrapper for requests that claims to prevent SSRF https://github.com/JordanMilne/Advocate.